package.json

```
@@ -752,11 +752,53 @@
     "viewsWelcome": [
       {
         "view": "terraform.providers",
-        "contents": "The active editor cannot provide information about installed providers. [Learn more about providers](https://www.terraform.io/docs/language/providers/index.html). You may need to update your language server version."
+        "contents": "This view requires the language server to be enabled:\n[Open Settings](command:terraform.openSettingsJson)",
+        "when": "config.terraform.languageServer.enable === false"
+      },
+      {
+        "view": "terraform.providers",
+        "contents": "There are no open Terraform files. Please open a Terraform configuration file to see installed providers.",
+        "when": "config.terraform.languageServer.enable && terraform.providers.documentOpened === false"
+      },
+      {
+        "view": "terraform.providers",
+        "contents": "The active document is not a Terraform file. Please open a Terraform configuration file to see installed providers.",
+        "when": "config.terraform.languageServer.enable && terraform.providers.documentIsTerraform === false"
+      },
+      {
+        "view": "terraform.providers",
+        "contents": "There are no installed providers found for the current open file.\n[Learn more about providers](https://www.terraform.io/docs/language/providers/index.html)",
+        "when": "config.terraform.languageServer.enable && terraform.providers.noProviders === true"
+      },
+      {
+        "view": "terraform.providers",
+        "contents": "The active editor cannot provide information about installed providers.\n[Learn more about providers](https://www.terraform.io/docs/language/providers/index.html)",
+        "when": "config.terraform.languageServer.enable && terraform.providers.noResponse === true"
+      },
+      {
+        "view": "terraform.modules",
+        "contents": "This view requires the language server to be enabled:\n[Open Settings](command:terraform.openSettingsJson)",
+        "when": "config.languageServer.enable === false"
```

There are still 5 `terraform.` prefixes missing ;)
```suggestion
        "when": "config.terraform.languageServer.enable === false"
```

---

One thing at a time! :P